doc/guides/primordials.md

```diff
@@ -0,0 +1,500 @@
+
+```js
+const arrayCopy = ArrayPrototypeSlice(array);
+ReflectApply(func, null, array);
+```
+
+</details>
+
+<details>
+
+<summary><code>%Object.fromEntries%</code> iterate over an array</summary>
+
+```js
+{
+  // Unsafe code example:
+  // 1. Lookup @@iterator property on `array` (user-mutable if user provided).
+  // 2. Lookup @@iterator property on %Array.prototype% (user-mutable).
+  // 3. Lookup `next` property on %ArrayIteratorPrototype% (user-mutable).
+  const obj = ObjectFromEntries(array);
+}
+
+{
+  // Safe example using `SafeArrayIterator`:
+  const obj = ObjectFromEntries(new SafeArrayIterator(array));
+}
+
+{
+  // Safe example without using `SafeArrayIterator`:
+  const obj = {};
+  for (let i = 0; i < array.length; i++) obj[array[i][0]] = array[i][1];
+  // In a hot code path, this would be the preferred method.
+}
+```
+
+</details>
+
+<details>
+
+<summary><code>%Promise.all%</code>,
+         <code>%Promise.allSettled%</code>,
+         <code>%Promise.any%</code>, and
+         <code>%Promise.race%</code> iterate over an array</summary>
+
+```js
+// 1. Lookup @@iterator property on `array` (user-mutable if user provided).
+// 2. Lookup @@iterator property on %Array.prototype% (user-mutable).
+// 3. Lookup `next` property on %ArrayIteratorPrototype% (user-mutable).
+PromiseAll(array); // unsafe
+
+PromiseAll(new SafeArrayIterator(array)); // safe
+```
+
+</details>
+
+<details>
+
+<summary><code>%Map%</code>, <code>%Set%</code>, <code>%WeakMap%</code>, and
+         <code>%WeakSet%</code> constructors iterate over an array</summary>
+
+```js
+// 1. Lookup @@iterator property on %Array.prototype% (user-mutable).
+// 2. Lookup `next` property on %ArrayIteratorPrototype% (user-mutable).
+const set = new SafeSet([1, 2, 3]);
+```
+
+```js
+const set = new SafeSet();
+set.add(1).add(2).add(3);
+```
+
+</details>
+
+### Promise objects
+
+<details>
+
+<summary><code>%Promise.prototype.finally%</code> looks up <code>then</code>
+         property of the Promise instance</summary>
+
+```js
+// User-land
+Promise.prototype.then = function then(a, b) {
+  return Promise.resolve();
+};
+
+// Core
+let finallyBlockExecuted = false;
+PromisePrototypeFinally(somePromiseThatEventuallySettles,
+                        () => { finallyBlockExecuted = true; });
+process.on('exit', () => console.log(finallyBlockExecuted)); // false
+```
+
+```js
+// User-land
+Promise.prototype.then = function then(a, b) {
+  return Promise.resolve();
+};
+
+// Core
+let finallyBlockExecuted = false;
+(async () => {
+  try {
+    return await somePromiseThatEventuallySettles;
+  } finally {
+    finallyBlockExecuted = true;
+  }
+})();
+process.on('exit', () => console.log(finallyBlockExecuted)); // true
+```
+
+</details>
+
+### (Async) Generator functions
+
+Generator and async generators returned by generator functions and async
+generator functions are relying on user-mutable, their use in core should be
```

Comment:
Shouldn't something come after `user-mutable`?  Also, a semicolon is more grammatically correct here than a comma.
```suggestion
generator functions are relying on user-mutable ???; their use in core should be
```